### PR TITLE
Externalize timezone for CRON evaluation as config property.

### PR DESCRIPTION
**Pull request by duftler**
- Default to America/Los_Angeles.
  - Respect timezone when determining if triggers and action instance are in sync.

### CODE REVIEW

**Comment by duftler** [comment]
@cfieber please review.

File: echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineTriggerConverter.groovy [hunk 1]
```diff
@@ -51,26 +52,27 @@ class PipelineTriggerConverter {
...
   }
 
-  static boolean isInSync(ActionInstance actionInstance, Trigger trigger) {
+  static boolean isInSync(ActionInstance actionInstance, Trigger trigger, String timeZoneId) {
     if (trigger.type == Trigger.Type.CRON.toString()) {
       return (
         actionInstance.trigger instanceof CronTrigger &&
-          trigger.cronExpression == ((CronTrigger) actionInstance.trigger).cronExpression
+          trigger.cronExpression == ((CronTrigger) actionInstance.trigger).cronExpression &&
+          timeZoneId == actionInstance?.context?.parameters?.triggerTimeZoneId
```
Comment (cfieber) [inline echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineTriggerConverter.groovy]: I guess in the case of existing triggers where timeZoneId doesn't explicitly exist, they will just get updated here? (isInSync will be false the first time it is checked, which will re-persist the trigger including the configured timezone)

Also applies in the case of a persisted trigger with a timezone, and changing the system configured timezone?

File: echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineConfigsPollingAgent.groovy [hunk 2]
```diff
@@ -46,6 +46,9 @@ class PipelineConfigsPollingAgent extends AbstractPollingAgent {
   private final long intervalMs
   private final ActionsOperator actionsOperator
 
+  @Value('${scheduler.cron.timezone:America/Los_Angeles}')
```
Comment (cfieber) [inline echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineConfigsPollingAgent.groovy]: Could you add a constructor arg and put the `@Value` annotation there, and make the field final

**Comment by cfieber** [comment]
nice, thanks LGTM

File: echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineTriggerConverter.groovy (hunk 1 above)
Comment (duftler) [inline echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineTriggerConverter.groovy]: Yup, exactly.
I verified this behavior with a running system as well.

File: echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineConfigsPollingAgent.groovy (hunk 2 above)
Comment (duftler) [inline echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineConfigsPollingAgent.groovy]: Done (sorry, hadn't spotted the existing `@Value` there).

**Pull request merged**